pandas/tests/groupby/test_groupby.py

```diff
@@ -134,6 +134,34 @@ def func(dataf):
     result = df.groupby("X", squeeze=False).count()
     assert isinstance(result, DataFrame)
 
+    # https://github.com/pandas-dev/pandas/issues/28330
+    # Test groupby operations on subclassed dataframes/series
+    class ChildSeries(Series):
```

Comment:
We have these in `pandas/util/testing.py` You can reuse them.

Reply:
Great!